frontend/src/modules/controllers/AlarmModal.tsx

```diff
@@ -44,11 +48,39 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));
 
+/**
+ * @typedef AlarmAdjustProps
+ *
+ * Interface for the alarm adjustments
+ *
+ * @prop {number} committedMin Lower Set Alarm Range Value
+ * @prop {number} committedMax Upper Set Alarm Range Value
+ */
 export interface AlarmAdjustProps {
   committedMin: number;
   committedMax: number;
 }
 
+/**
+ * @typedef Props
+ *
+ * Props interface for the AlarmModal component
+ *
+ * @prop {string} label Alarm label to display in UI
+ * @prop {string} units Alarm paramater unit measurement to display
+ * @prop {number} committedMin Lower Set Alarm Range Value
+ * @prop {number} committedMax Upper Set Alarm Range Value
+ * @prop {boolean} disableAlarmButton Toggle to show/hide alarm button
+ * @prop {function} updateModalStatus Callback to send current modal open/close status
+ * @prop {function} onModalClose Callback on Modal close
+ * @prop {function} requestCommitRange Callback on updating the Alarm range values
+ * @prop {string} stateKey Unique identifier of alarm (eg spo2, fio2...)
+ * @prop {number} step Alarm step difference between Range (Defaults to 1)
+ * @prop {boolean} openModal Default value to toggle Open/Close Alarm Modal
+ * @prop {boolean} contentOnly Switch to display only Alarm Modal content vs Alarm Modal
+ * @prop {boolean} labelHeading Switch to show/hide Alarm Label inside Alarm Modal content
+ * @prop {number[]} alarmRangeValues Alarm Range Values
```

Comment:
This is Alarm Range values with which Alarm modal initializes with. If its empty it would show up default committedMin and committedMax as values in UI. For example, If FiO2 current alarm range values are [20-80], component would pass those values through `alarmRangeValues `